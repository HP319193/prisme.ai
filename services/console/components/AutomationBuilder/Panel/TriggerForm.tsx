import { useTranslation } from 'next-i18next';
import { FC, useMemo } from 'react';
import { Schema, SchemaForm } from '@prisme.ai/design-system';
import useSchema from '../../SchemaForm/useSchema';
import { useWorkspace } from '../../WorkspaceProvider';
import { useApps } from '../../AppsProvider';

interface TriggerFormProps {
  trigger?: Prismeai.When;
  onChange: (v: Prismeai.When) => void;
}

const buttons: any[] = [];

export const TriggerForm: FC<TriggerFormProps> = ({ trigger, onChange }) => {
  const { t } = useTranslation('workspaces');

  const { workspace } = useWorkspace();
  const { appInstances } = useApps();
  const { extractAutocompleteOptions } = useSchema({
    config: workspace.config,
    apps: appInstances.get(workspace.id),
    workspace,
  });

  const initialValue = useMemo(
    () => ({
      events: trigger?.events || [],
      endpoint: trigger?.endpoint ?? false,
    }),
    [trigger]
  );
  const schema: Schema = useMemo(
    () => ({
      type: 'object',
      properties: {
        events: {
          type: 'array',
          title: t('automations.trigger.events.title'),
          description: t('automations.trigger.events.help'),
          items: {
            type: 'string',
            title: t('automations.trigger.events.item'),
            'ui:widget': 'autocomplete',
            'ui:options': {
              autocomplete: 'events:emit',
            },
          },
        },
        // dates: {
        //   title: t('automations.trigger.dates.title'),
        //   description: t('automations.trigger.dates.help'),
        // },
        endpoint: {
          type: 'boolean',
          title: t('automations.trigger.endpoint.custom'),
          description: t('automations.trigger.endpoint.help'),
        },
      },
    }),
    [t]
  );

  return (
    <div className="overflow-x-auto">
      <SchemaForm
        schema={schema}
        initialValues={initialValue}
        onChange={onChange}
        locales={{
          addItem: t('automations.trigger.events.add'),
        }}
        buttons={buttons}
        utils={{ extractAutocompleteOptions }}
      />
    </div>
  );
};

export default TriggerForm;
