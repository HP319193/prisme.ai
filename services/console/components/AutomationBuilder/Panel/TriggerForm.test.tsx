import TriggerForm from './TriggerForm';
import renderer from 'react-test-renderer';

it('should render', () => {
  const trigger = {
    events: ['foo'],
  };
  const onSubmit = jest.fn();
  const root = renderer.create(
    <TriggerForm trigger={trigger} onSubmit={onSubmit} />
  );
  expect(root.toJSON()).toMatchSnapshot();
});
