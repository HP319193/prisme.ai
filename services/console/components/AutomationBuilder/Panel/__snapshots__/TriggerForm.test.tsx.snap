// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <fieldset
      className="p-fieldset p-component pt-5 mb-4"
      id={null}
      onClick={null}
      style={null}
    >
      <legend
        className="p-fieldset-legend p-unselectable-text"
        onClick={[Function]}
      >
        <span
          className="p-fieldset-legend-text"
          id="pr_id_1_header"
        >
          automations.trigger.events.title
        </span>
      </legend>
      <div
        aria-hidden={false}
        aria-labelledby="pr_id_1_header"
        className="p-toggleable-content"
        id="pr_id_1_content"
        role="region"
      >
        <div
          className="p-fieldset-content"
        >
          <div
            className="p-field mb-5"
          >
            <span
              className="flex p-float-label mx-2 undefined"
            >
              <div
                className="p-chips p-component p-inputwrapper p-inputwrapper-filled"
                id={null}
                style={null}
              >
                <ul
                  className="p-inputtext p-chips-multiple-container"
                  onClick={[Function]}
                >
                  <li
                    className="p-chips-token p-highlight"
                  >
                    <span
                      className="p-chips-token-label"
                    >
                      foo
                    </span>
                    <span
                      className="p-chips-token-icon pi pi-times-circle"
                      onClick={[Function]}
                    />
                  </li>
                  <li
                    className="p-chips-input-token"
                  >
                    <input
                      aria-labelledby={null}
                      disabled={null}
                      name="events"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder={null}
                      readOnly={false}
                      type="text"
                    />
                  </li>
                </ul>
              </div>
              <label
                htmlFor="events"
              />
            </span>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      className="p-fieldset p-component pt-5 mb-4"
      id={null}
      onClick={null}
      style={null}
    >
      <legend
        className="p-fieldset-legend p-unselectable-text"
        onClick={[Function]}
      >
        <span
          className="p-fieldset-legend-text"
          id="pr_id_2_header"
        >
          automations.trigger.dates.title
        </span>
      </legend>
      <div
        aria-hidden={false}
        aria-labelledby="pr_id_2_header"
        className="p-toggleable-content"
        id="pr_id_2_content"
        role="region"
      >
        <div
          className="p-fieldset-content"
        >
          automations.trigger.dates.help
        </div>
      </div>
    </fieldset>
    <fieldset
      className="p-fieldset p-component pt-5 mb-4"
      id={null}
      onClick={null}
      style={null}
    >
      <legend
        className="p-fieldset-legend p-unselectable-text"
        onClick={[Function]}
      >
        <span
          className="p-fieldset-legend-text"
          id="pr_id_3_header"
        >
          automations.trigger.endpoint.title
        </span>
      </legend>
      <div
        aria-hidden={false}
        aria-labelledby="pr_id_3_header"
        className="p-toggleable-content"
        id="pr_id_3_content"
        role="region"
      >
        <div
          className="p-fieldset-content"
        >
          <div
            className="p-field mb-5"
          >
            <label
              className="mx-2 flex flex-1 align-center flex-col"
            >
              <div
                aria-checked={false}
                className="p-inputswitch p-component mr-2"
                id={null}
                onClick={[Function]}
                role="checkbox"
                style={null}
              >
                <div
                  className="p-hidden-accessible"
                >
                  <input
                    aria-checked={false}
                    aria-labelledby={null}
                    checked={false}
                    disabled={false}
                    id={null}
                    name={null}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    role="switch"
                    type="checkbox"
                  />
                </div>
                <span
                  className="p-inputswitch-slider"
                />
              </div>
              automations.trigger.endpoint.custom
            </label>
          </div>
        </div>
      </div>
    </fieldset>
    <button
      className="p-button p-component"
      disabled={false}
    >
      automations.trigger.save
    </button>
  </form>
</div>
`;
